Specs/Core/EllipsoidRhumbLineSpec.js

```diff
@@ -0,0 +1,581 @@
+        var ellipsoid = Ellipsoid.WGS84;
+        var fifteenDegrees = Math.PI / 12;
+        var initial = new Cartographic(fifteenDegrees, fifteenDegrees);
+        var distance = ellipsoid.maximumRadius * fifteenDegrees;
+
+        var oneDegree = CesiumMath.RADIANS_PER_DEGREE;
+        var eightyNineDegrees = 89 * oneDegree;
+        var eightyNinePointNineDegrees = 89.9 * oneDegree;
+        var ninetyDegrees = 90 * oneDegree;
+        var ninetyPointOneDegrees = 90.1 * oneDegree;
+        var ninetyPointZeroTwoDegrees = 90.02 * oneDegree;
+
+        var rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, eightyNineDegrees, distance, ellipsoid);
+        var rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON3);
+
+        rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, eightyNinePointNineDegrees, distance, ellipsoid);
+        rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON3);
+
+        rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, ninetyDegrees, distance, ellipsoid);
+        rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON3);
+
+        rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, ninetyPointOneDegrees, distance, ellipsoid);
+        rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON3);
+
+        rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, ninetyPointZeroTwoDegrees, distance, ellipsoid);
+        rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON3);
+    });
+
+    it('test sphereoid across meridian', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var fifteenDegrees = Math.PI / 12;
+        var initial = new Cartographic(-fifteenDegrees, 0.0);
+        var final = new Cartographic(fifteenDegrees, 0.0);
+        var distance = ellipsoid.maximumRadius * 2 * fifteenDegrees;
+
+        var rhumb1 = new EllipsoidRhumbLine(initial, final, ellipsoid);
+        var rhumb2 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, CesiumMath.PI_OVER_TWO, distance, ellipsoid);
```

Comment:
Is this spec along the equator but crossing the PM? If so, seems like it should be straightforward to compute the expected heading/distance values too since the WGS84 reference ellipsoid is circular about the equator.

Reply:
Right, that's why we do it with the distance above. For `rhumb1` the surface distance is calculated and must be equal to rhumb2, whose surface distance is provided.

So multiple checks are satisfied simultaneously.